pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -416,12 +418,30 @@ func (ctx *context) getDIDDocAndConnection(pubDID string) (*did.Doc, *Connection
 
 	logger.Debugf("creating new '%s' did for connection", didMethod)
 
+	// get the route configs (pass empty service endpoint, as default servie endpoint added in VDRI)
+	serviceEndpoint, routingKeys, err := route.GetRouterConfig(ctx.routeSvc, "")
+	if err != nil {
+		return nil, nil, fmt.Errorf("did doc - fetch router config : %w", err)
+	}
+
 	// by default use peer did
-	newDidDoc, err := ctx.vdriRegistry.Create(didMethod)
+	newDidDoc, err := ctx.vdriRegistry.Create(
+		didMethod,
+		vdri.WithServiceEndpoint(serviceEndpoint),
+		vdri.WithRoutingKeys(routingKeys),
+	)
 	if err != nil {
 		return nil, nil, fmt.Errorf("create %s did: %w", didMethod, err)
 	}
 
+	for _, svc := range newDidDoc.Service {
+		for _, recKey := range svc.RecipientKeys {
+			if err = route.AddKeyToRouter(ctx.routeSvc, recKey); err != nil {
```

Comment:
1. The [example](https://github.com/hyperledger/aries-rfcs/blob/master/features/0211-route-coordination/README.md#keylist-update) `keylist_update` message in the RFC suggests you can register several keys in a single batch
2. Need a TODO: it's possible that one or several of those recipient keys can be registered in another router.

Reply:
Yes, missed to create a TODO. Created a new issue and [added](https://github.com/hyperledger/aries-framework-go/compare/70e04d120f1905e8fee04847f9fa56eaa14711e8..488523d9b32e5efb07b85bcc1dd8b3965fc9a9e1#diff-7cc3df019152fe2c557826a312cc0a12R398) a TODO.